cluster-agent/controllers/argoproj.io/application_controller_test.go

```diff
@@ -0,0 +1,692 @@
+		var dbQueries db.AllDatabaseQueries
+		var guestbookApp *appv1.Application
+		var reconciler argo.ApplicationReconciler
+		var err error
+
+		BeforeEach(func() {
+			scheme, argocdNamespace, kubesystemNamespace, workspace, err := eventlooptypes.GenericTestSetup()
+			Expect(err).To(BeNil())
+
+			err = appv1.AddToScheme(scheme)
+			Expect(err).To(BeNil())
+
+			gitopsDepl := &managedgitopsv1alpha1.GitOpsDeployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "my-gitops-depl",
+					Namespace: workspace.Name,
+					UID:       uuid.NewUUID(),
+				},
+			}
+
+			// Fake kube client.
+			k8sClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(gitopsDepl, workspace, argocdNamespace, kubesystemNamespace).Build()
+
+			// Database Connection.
+			dbQueries, err = db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+
+			err = testSetup()
+			Expect(err).To(BeNil())
+
+			_, managedEnvironment, _, gitopsEngineInstance, _, err = createSampleData(dbQueries)
+			Expect(err).To(BeNil())
+
+			guestbookApp = &appv1.Application{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Application",
+					APIVersion: "argoproj.io/v1alpha1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+					Labels: map[string]string{
+						dbID: "test-my-application",
+					},
+				},
+				Spec: appv1.ApplicationSpec{
+					Source: appv1.ApplicationSource{
+						Path:           "guestbook",
+						TargetRevision: "HEAD",
+						RepoURL:        "https://github.com/argoproj/argocd-example-apps.git",
+					},
+					Destination: appv1.ApplicationDestination{
+						Namespace: "guestbook",
+						Server:    "https://kubernetes.default.svc",
+					},
+					Project: "default",
+				},
+				Status: appv1.ApplicationStatus{
+					Health: appv1.HealthStatus{
+						Status: "Healthy",
+					},
+					Sync: appv1.SyncStatus{
+						Status: "Synced",
+					},
+				},
+			}
+
+			reconciler = argo.ApplicationReconciler{
+				Client:        k8sClient,
+				Scheme:        scheme,
+				DB:            dbQueries,
+				TaskRetryLoop: sharedutil.NewTaskRetryLoop("application-reconciler"),
+			}
+		})
+
+		It("Create New Application CR in namespace and database and verify that ApplicationState is created", func() {
+			// Close database connection.
+			defer dbQueries.CloseDatabase()
+			defer testTeardown()
+
+			ctx = context.Background()
+
+			// Add databaseID label to applicationID.
+			databaseID := guestbookApp.Labels[dbID]
+			applicationDB := &db.Application{
+				Application_id:          databaseID,
+				Name:                    name,
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+
+			applicationState := &db.ApplicationState{
+				Applicationstate_application_id: applicationDB.Application_id,
+			}
+
+			// Create a new ArgoCD Application.
```

Comment:
Ginkgo actually has a neat way to include comments like these in the test output itself.

You can convert these explanatory comments in all your tests to: 
`By("...")`
for example, in this case, this comment would become
`By("Create a new Argo CD Application")`

Then this comment would be output during the test, which helps with test debugging.